### PR TITLE
Users can edit connection credentials

### PR DESCRIPTION
**Pull request by jferris**
* Adds edit links for each connection
* Implements edit/update for authentications

https://trello.com/c/xrYEiY3u

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -154,6 +155,10 @@ en:
         title: "Greenhouse"
         slogan: >
           Please fill out the following fields to connect to this application.
+      edit:
+        title: "Greenhouse"
```
Comment (base10) [inline config/locales/en.yml]: Thoughts on refactoring the duplication out between "new" and "edit" here? Potentially using `integration_id.name" instead?

**Comment by base10** [comment]
One question, otherwise LGTM

File: config/locales/en.yml (hunk 1 above)
Comment (jferris) [inline config/locales/en.yml]: I've been thinking we probably don't need any variation between these pages, and the extra translations are probably just noise.

What do you think about doing a separate pull request to:

* Use the integration title for all these titles
* Use the same slogan regardless of integration

We're already doing both of those, but we repeat the translations anyway.

Comment (base10) [inline config/locales/en.yml]: I like that idea.
